### PR TITLE
[SM] NoticeBox + MessageDetails

### PR DESCRIPTION
**Pull request by U-DON**
Closes:
- department-of-veterans-affairs/messaging-fe#29
- department-of-veterans-affairs/messaging-fe#43

Added NoticeBox at bottom of Thread view.
Added MessageDetails button and dialog that appears when button is clicked.
Threads are linked and accessible from folders (although, API only allows the first thread in Inbox at the moment).
## Mocks

Desktop: https://marvelapp.com/1bdd199/screen/15044678/layer/18718117
## Preview

screenshot from 2016-09-16 00 06 35

### CODE REVIEW

File: src/js/messaging/containers/Thread.jsx [hunk 1]
```diff
@@ -1,55 +1,62 @@
 import React from 'react';
 import { connect } from 'react-redux';
 
-import { fetchThread } from '../actions/messages';
+import { fetchThread, setVisibleDetails } from '../actions/messages';
 import Message from '../components/Message';
+import NoticeBox from '../components/NoticeBox';
 
 class Thread extends React.Component {
   componentWillMount() {
```
Comment (webinista) [inline src/js/messaging/containers/Thread.jsx]: In light of what we learned with #3066/#3069, should this be `componentDidMount`?

Comment (U-DON) [inline src/js/messaging/containers/Thread.jsx]: Yes, I can change that.

Comment (U-DON) [inline src/js/messaging/containers/Thread.jsx]: I'll change it everywhere we use it in messaging as well.

**Review by webinista — CHANGES_REQUESTED** [review]
LGTM, aside from the `componentDidMount` vs `componentWillMount` change.

**Pull request merged**